thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -2928,12 +3063,35 @@ class HashDedupSlaveActivityBase : public CSlaveActivity
                 lastEog = false;
                 if (bucketHandler->addRow(row)) // true if new, i.e. non-duplicate (does not take ownership)
                 {
-                    dataLinkIncrement();
-                    return row.getClear();
+                    if (!keepBest)
+                    {
+                        dataLinkIncrement();
+                        return row.getClear();
+                    }
                 }
             }
             else
             {
+                if (keepBest && !keepBestRowsReadyToReturn)
+                {
+                    bucketHandler->clearCallbacks();
+                    // This should cause one of the buckets to spill (used for testing)
+                    if (testSpillTimes)
+                    {
+                        bucketHandler->spillBucket(false);
+                        testSpillTimes--;
+                    }
```

Comment:
I/we realize now, this doesn't test spilling at expected.. as this is at end-of-stream. It still tests some of the mechanisms though, as it will cause a new bucket handler to come into existence with a new set of subdivided buckets, so it may be good enough, particularly as it's tricky to cause the fake test midstream efficiently.